R/make_clean_names.R

```diff
@@ -24,58 +51,91 @@
 #' # not run:
 #' # make_clean_names(names(poorly_named_df))
-#' 
-make_clean_names <- function(string, case = c(
-  "snake", "lower_camel", "upper_camel", "screaming_snake",
-  "lower_upper", "upper_lower", "all_caps", "small_camel",
-  "big_camel", "old_janitor", "parsed", "mixed", "none"
-)) {
+#'
+#' @importFrom stringi stri_trans_general
+#' @importFrom stringr str_replace str_replace_all
+make_clean_names <- function(string,
+                             case = c(
+                               "snake", "lower_camel", "upper_camel", "screaming_snake",
+                               "lower_upper", "upper_lower", "all_caps", "small_camel",
+                               "big_camel", "old_janitor", "parsed", "mixed", "none"
+                             ),
+                             replace=
+                               c(
+                                 "'"="",
+                                 "\""="",
+                                 "%"=".percent_",
+                                 "#"=".number_"
+                               ),
+                             ascii=TRUE) {
   
-  # old behavior, to provide easy fix for people whose code breaks with the snakecase integration
+  # Handling "old_janitor" case for backward compatibility
   case <- match.arg(case)
   if (case == "old_janitor") {
     return(old_make_clean_names(string))
   }
-  
-  ### new behaviour with snakecase integration
-  # Takes a data.frame, returns the same data frame with cleaned names
-  old_names <- string
-  new_names <- old_names %>%
-    gsub("'", "", .) %>% # remove single quotation marks
-    gsub("\"", "", .) %>% # remove double quotation marks
-    gsub("%", ".percent_", .) %>% # starting with "." as a workaround, to make
-    # ".percent" a valid name. The "." will be replaced in the call to to_any_case
-    # via the preprocess argument anyway.
-    gsub("#", ".number_", .) %>%
-    gsub("^[[:space:][:punct:]]+", "", .) %>% # remove leading spaces & punctuation
-    make.names(.) %>%
-    # Handle dots, multiple underscores, case conversion, string transliteration
-    # Parsing option 4 removes underscores around numbers, #153
-    snakecase::to_any_case(.,
-      case = case, sep_in = "\\.",
-      transliterations = c("Latin-ASCII"), parsing_option = 1,
+
+  replaced_names <-
+    stringr::str_replace_all(
+      str=string,
+      pattern=replace
+    )
+  transliterated_names <-
+    if (ascii) {
+      stringi::stri_trans_general(
+        replaced_names,
+        id="Greek-Latin;Latin-ASCII;Accents-Any;Any-ASCII"
+      )
+    } else {
+      replaced_names
+    }
+  # Remove starting spaces and punctuation
+  good_start <-
+    stringr::str_replace(
+      str=transliterated_names,
+      pattern="\\A[\\h\\s\\p{Punctuation}\\p{Symbol}\\p{Separator}\\p{Other}]*(.*)$",
+      replacement="\\1"
+    )
+  # make.names() is dependent on the locale and therefore will return different
+  # system-dependent values.
```

Comment:
Do you have a sense of, in practice, what the most common locale inconsistency in `make.names` might be?  Just wondering.

Reply:
The issue in #268 is one example.  I think that it mostly relates to Unicode characters that are local letters.